Automattic/Gravatar-SDK-iOS
# Support for oauth callback https for iOS 17.4 +

Closes #554 

### Description

This PR implements the new `ASWebAuthenticationSession` API for callback URLs with `https` schemes.

For users, it's a non-breaking change, and if they support only iOS 17.4+, they won't need to implement the app/scene delegate auth callback.

I also made some changes, introducing `SessionStorage` to silence some concurrency warnings (Swift 6 errors). After trying many different paths, this one the only one that worked with localized changes (not spreading to other classes).

If you want to check the warning and experiment with other solutions, roll back the last 3 commits from this branch

### Testing Steps

- Log out from the Quick Editor.
- Open the Quick Editor authorisation.
- Check that `Approbe` and `Deny` buttons on the web view work.
  - You can also check that `_ = await OAuthSession.handleCallback(url)` in `SceneDelegate.swift:55` is not called.
- To test pre iOS17.4, comment the `if #available(iOS 17.4, *)` path and run again.
  - Check that the Auth web view works as expected
  - Check that `_ = await OAuthSession.handleCallback(url)` in `SceneDelegate.swift:55` is called.
